apps/prairielearn/elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -552,7 +578,8 @@ def render(element_html: str, data: pl.QuestionData) -> str:
                 "indent": (attempt["indent"] or 0) * TAB_SIZE_PX,
                 "badge_type": attempt.get("badge_type", ""),
                 "icon": attempt.get("icon", ""),
-                "distractor_feedback": attempt.get("distractor_feedback", ""),
+                "ordering_feedback": attempt.get("ordering_feedback", ""),
+                "is_disordered": attempt.get("is_disordered", False),
```

Comment:
do you need to have the `is_disordered` property here, or can you base all the logic on whether or not `ordering_feedback` exists?

Reply:
I will improve this part

Reply:
I have tried to improve it as you said, but I have bumped into a issue. Right now I have code like:
```
for i, attempt in enumerate(student_answer):
    if i not in misordered_blocks:
        attempt["ordering_feedback"] = ""
```
This means only misordered blocks keep their ordering_feedback, and we should be able to base all the logic on whether or not ordering_feedback exists. The problem is — once a block is correct, its feedback gets cleared, and since we’re relying on student_answer to store it, that feedback is gone. After a resubmission, we can’t recover it.

To fix this, we’d need a more stable source that stores the original feedback and lets us reapply it each time. I'm not sure what the best source would be, and I’m a little concerned about increasing memory use. Another option could be keeping a flag like is_disordered to explicitly track misordered blocks — and if we go that route, I’d be happy to rename it to something like is_misordered for clarity.

Just wanted to see if that makes sense, or if there’s a better approach I might be missing. Thanks!

Reply:
> After a resubmission, we can’t recover it.

I don't think this is true. After each submission, during the `parse` phase, the `ordering_feedback` is accessed from `data['correct_answers']`. The properties you put here only affect the rendering of the current submission, and have nothing to do with how future submissions are rendered. 

I think you can just make this change here: and it will solve the problem

```suggestion
                "ordering_feedback":  attempt.get("ordering_feedback", "") if attempt.get("is_disordered", False) else ""
```

Reply:
I’ve made some changes and fixed a few small errors. In the updated version, I also added a ValueError to ensure that the ordering-feedback attribute can only be used on blocks where correct=true. Since we already have distractor-feedback for blocks correct=false.